src/Symfony/Component/Form/Extension/Validator/EventListener/ValidationListener.php

```diff
@@ -35,17 +40,27 @@ public static function getSubscribedEvents()
         return [FormEvents::POST_SUBMIT => 'validateForm'];
     }
 
-    public function __construct(ValidatorInterface $validator, ViolationMapperInterface $violationMapper)
+    public function __construct(ValidatorInterface $validator, ViolationMapperInterface $violationMapper, ?EventDispatcherInterface $eventDispatcher = null)
     {
         $this->validator = $validator;
         $this->violationMapper = $violationMapper;
+
+        if (!$this->eventDispatcher) {
+            @trigger_error(sprintf('The "$eventDispatcher" argument to the "%s" constructor will be required in Symfony 6.0.', self::class));
+        }
+
+        $this->eventDispatcher = $eventDispatcher;
     }
 
     public function validateForm(FormEvent $event)
     {
         $form = $event->getForm();
 
         if ($form->isRoot()) {
+            if ($this->eventDispatcher) {
+                $this->eventDispatcher->dispatch(new PreValidateEvent($event->getForm(), $event->getData()), 'form.pre_validate');
```

Comment:
can't you use the constant instead?

Reply:
That was leftover from before the constants. Thanks for flagging it.